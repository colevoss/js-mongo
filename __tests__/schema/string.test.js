const Schema = require('../../src/Schema');

describe('Schema.string', () => {
  test('Returns valid string', () => {
    const testData = { test: 'test string' };

    expect(Schema.string('test', testData)).toBe(testData.test);
  });

  test('Returns error', () => {
    const testData = { test: 1 };

    const checked = Schema.string('test', testData);

    expect(checked).toBeInstanceOf(TypeError);
    expect(checked).toMatchSnapshot();
  });

  test('__type', () => {
    expect(Schema.string.__type).toBe('string');
  });
});
