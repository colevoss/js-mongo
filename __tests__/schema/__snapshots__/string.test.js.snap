// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema.string Returns error 1`] = `[TypeError: Expected test to be of type string but recieved number of 1]`;
